lib/matplotlib/backends/__init__.py

```diff
@@ -43,50 +44,22 @@ def pylab_setup(name=None):
     '''
     # Import the requested backend into a generic module object
     if name is None:
-        # validates, to match all_backends
         name = matplotlib.get_backend()
-    if name.startswith('module://'):
-        backend_name = name[9:]
-    else:
-        backend_name = 'backend_' + name
-        backend_name = backend_name.lower()  # until we banish mixed case
-        backend_name = 'matplotlib.backends.%s' % backend_name.lower()
-
-    # the last argument is specifies whether to use absolute or relative
-    # imports. 0 means only perform absolute imports.
-    backend_mod = __import__(backend_name, globals(), locals(),
-                             [backend_name], 0)
-
-    # Things we pull in from all backends
-    new_figure_manager = backend_mod.new_figure_manager
-
-    # image backends like pdf, agg or svg do not need to do anything
-    # for "show" or "draw_if_interactive", so if they are not defined
-    # by the backend, just do nothing
-    def do_nothing_show(*args, **kwargs):
-        frame = inspect.currentframe()
-        fname = frame.f_back.f_code.co_filename
-        if fname in ('<stdin>', '<ipython console>'):
-            warnings.warn("""
-Your currently selected backend, '%s' does not support show().
-Please select a GUI backend in your matplotlibrc file ('%s')
-or with matplotlib.use()""" %
-                          (name, matplotlib.matplotlib_fname()))
-
-    def do_nothing(*args, **kwargs):
-        pass
-
-    backend_version = getattr(backend_mod, 'backend_version', 'unknown')
-
-    show = getattr(backend_mod, 'show', do_nothing_show)
-
-    draw_if_interactive = getattr(backend_mod, 'draw_if_interactive',
-                                  do_nothing)
-
-    _log.debug('backend %s version %s', name, backend_version)
+    backend_name = (name[9:] if name.startswith("module://")
+                    else "matplotlib.backends.backend_{}".format(name.lower()))
+    backend_mod = importlib.import_module(backend_name)
+    # Create a local Backend class whose body corresponds to the contents of
+    # the backend module.  This allows the Backend class to fill in the missing
+    # methods through inheritance.
+    Backend = type("Backend", (_Backend,), vars(backend_mod))
```

Comment:
Indeed, backends that were using `@_Backend.export` don't really need that anymore (the contents of the local _Backend class could just be dumped at the module level (or alternatively they could just do without the temporary creation of a new Backend class))... except for one thing: the _Backend class still allows inheritance between backends, so that e.g. _BackendQt5Agg can inherit from _BackendQt5.